src/pool/srv_pool.c

```diff
@@ -5303,6 +5303,101 @@ ds_pool_extend_handler(crt_rpc_t *rpc)
 	crt_reply_send(rpc);
 }
 
+static int
+pool_discard(crt_context_t ctx, struct pool_svc *svc, struct pool_target_addr_list *list)
+{
+	struct pool_tgt_discard_in	*ptdi_in;
+	struct pool_tgt_discard_out	*ptdi_out;
+	struct rdb_tx			tx;
+	struct pool_map			*map;
+	crt_rpc_t			*rpc;
+	d_rank_list_t			*rank_list = NULL;
+	crt_opcode_t			opc;
+	int				i;
+	int				nr = 0;
+	int				rc;
+
+	rc = rdb_tx_begin(svc->ps_rsvc.s_db, svc->ps_rsvc.s_term, &tx);
+	if (rc != 0)
+		return rc;
+
+	ABT_rwlock_wrlock(svc->ps_lock);
```

Comment:
why need write lock? seems rd lock is enough as it only read the map?

Reply:
right, read lock is enough. than